src/features/defi/providers/univ2/components/UniV2Manager/Deposit/components/Approve.tsx

```diff
@@ -189,11 +211,11 @@ export const Approve: React.FC<FoxEthLpApproveProps> = ({ accountId, onNext }) =
       loading={state.loading}
       loadingText={translate('common.approveOnWallet')}
       preFooter={preFooter}
-      providerIcon={foxAsset?.icon}
+      providerIcon={asset1?.icon}
       learnMoreLink='https://shapeshift.zendesk.com/hc/en-us/articles/360018501700'
       onCancel={() => onNext(DefiStep.Info)}
       onConfirm={handleApprove}
-      contractAddress={FOX_TOKEN_CONTRACT_ADDRESS}
+      contractAddress={asset1ContractAddress}
```

Comment:
just thinking - the contract address shouldn't really be specific to the underlying assets

Reply:
For approvals we'll definitely need the contract addresses, though as you rightfully noted this won't handle the case where both assets 0 and 1 are *actual* ERC-20s - once we get data in with such pairs, we'll need to support multi-approvals both from an abstraction/logic perspective as well as having proper UX for it